File: miniJobs/UI/minijobs_admin/lib/models/job/job_application.dart

import 'package:json_annotation/json_annotation.dart';
import 'package:minijobs_admin/models/job/job.dart';

import '../../enumerations/job_application_status.dart';
import '../rating.dart';

part 'job_application.g.dart';
@JsonSerializable()
class JobApplication {
  int? id;
  JobApplicationStatus? status;
  int? jobId;
  DateTime? created;
  int? createdBy;
  Job? job;
  bool hasRated = false;
  Rating? rating;
  String? createdByName;
  String? jobName;
  JobApplication();
  JobApplication.withData(this.id, this.status, this.jobId,
      this.created,this.createdBy);

  factory JobApplication.fromJson(Map<String, dynamic> json) => _$JobApplicationFromJson(json);

  Map<String, dynamic> toJson() => _$JobApplicationToJson(this);
}

File: miniJobs/Domain/Entities/JobApplication.cs
using Domain.Common;
using Domain.Enums;
using Domain.Interfaces;

namespace Domain.Entities;

[Table("job_applications")]
public class JobApplication : BaseAuditableEntity, IEntity<int>
{
    [Key]
    [Column("id")]
    public int Id { get; set; }

    [Column("job_id")]
    public int JobId { get; set; }

    [Column("status")]
    public JobApplicationStatus Status { get; set; }

    [Column("is_deleted")]
    public bool IsDeleted { get; set; }

    [NotMapped]
    public Job Job { get; set; }

    [NotMapped]
    public bool HasRated { get; set; }

    [NotMapped]
    public string CreatedByName { get; set; }

    [NotMapped]
    public Rating Rating { get; set; }
}